### ISSUE TITLE
memoise does not cache in Shiny server

### ISSUE THREAD

**Issue by obkhan** [r-lib/memoise#37]
Hi,

I tried to get memoise to work with Shiny to cache results from a SPARQL endpoint to no avail.

Please see the following thread for more details:

https://groups.google.com/forum/#!topic/shiny-discuss/v4GLd5OSEvo

**Comment by jimhester** [comment]
Could you provide a minimal reproducible and complete example of the issue, ideally without the SPARQL dependency? Is this behavior seen with Shiny Server Pro or open source Shiny Server?

**Comment by oganm** [comment]
You can save a memoised object and load it when the app starts. If you want memory to be persistent within sessions, save the object with `saveRDS` or something similar at `onSessionEnded`. Be vary of overgrown caches.

**Comment by nzbart** [comment]
@oganm I'm trying to resolve a similar problem to @obkhan, and it appears that we both assumed that memoise using the file system cache should persist across sessions, processes, and even machines (via file share / drop-box). I'm not sure how `saveRDS` would solve the problem, short of using it instead of memoise?

**Comment by oganm** [comment]
At a lower level I have no idea how it works at a lower level but try

```r
library(memoise)
library(dplyr)

random = function(a){
    sample(1:10000000)
    return(a)
}

system.time(sapply(1:10,random)) # execution time of the function

memoRandom = memoise(random)
system.time(sapply(1:10,memoRandom))

saveRDS(memoRandom,file = 'memoRandom.rds')
```

Now switch to a different session and load the rds

```r
loadedRandom = readRDS('memoRandom.rds')
system.time(sapply(1:10,loadedRandom))
```

I don't know what is happening behind the scenes here as `object.size(memoRandom)` always remains the same but you'll see the size of the RDS file is increasing if you save after running memoRandom with more variables.

Try

```
sapply(1:100,memoRandom)
saveRDS(memoRandom,file = 'memoRandom2.rds')
```
and compare the file sizes

**Comment by nzbart** [comment]
@oganm An interesting approach. Maybe the memory cache is attached to the object that is saved via `saveRDS`, which would explain the increase in size? It's interesting that `saveRDS` must save more than `object.size` takes into account.

I've tried the latest version of the master branch, and this commit appears to have resolved the problem that I was having: 611cfadba3fb5198b4620d827f2984874196d181. I can now share file-system caches between different executions in different contexts.
